com.unity.netcode.gameobjects/Runtime/Messaging/NetworkContext.cs

```diff
@@ -0,0 +1,10 @@
+namespace Unity.Netcode
+{
+    public ref struct NetworkContext
+    {
+        public object SystemOwner;
```

Comment:
if we were to kill `SystemOwner` & `Bind` etc. (our own custom DI framework), would we need this `NetworkContext` struct at all?
I feel like we can just pass these values as params and that'd be one less type (one less thing to know about) during development.
I think cognitive load of lots of different small struct types won't worth it — same goes for `MessageHeader` but maybe that's a different story.

Reply:
We can't kill SystemOwner. The message handlers still need to be able to access NetworkManager. SystemOwner is a NetworkManager and its main use is to be passed to the handlers - it's an `object` because Matt requested (and I agree) that MessagingSystem should not have a hard dependency on NetworkManager - loose coupling.

Reply:
seeing 2 small structs (MessageHeader & NetworkContext) with a few params makes me wonder whether there could be 1 struct or just some params on APIs instead. if you think these structs are actually necessary then sure, let's keep them. I just don't feel like it but I probably have limited context and understanding of them atm (as I didn't write too many IMessages yet).

Reply:
MessageHeader is sent across the wire... NetworkContext is not. That's why MessageHeader is a property of NetworkContext... otherwise I'd have to duplicate everything from MessageHeader into NetworkContext. 2 small structs is a matter of DRY (something I'm usually not very good at, heh)

Reply:
I would like to revisit the conversation on `SystemOwner` and `IgnoreMessageWhenSystemOwnerNotOfType` at a later date, please? It's internal so it can be changed at any time. I want to get this merged so can we figure out what we want to do with this after feature freeze? It works for now.